### PR TITLE
refactor(app): separate epoch change executor logic and tests

### PR DESCRIPTION
**Pull request by snormore**
This PR moves the epoch change-specific executor logic and tests to their own sub-modules. It does **not** change any functionality or any of the code that was moved. It's just some prep for work I'm doing in the area.

### CODE REVIEW

File: core/application/src/tests.rs
Comment (ozwaldorf) [inline core/application/src/tests.rs:1]: Minor nit, but should we move the remaining things to `tests/mod.rs` as well?

**Comment by ozwaldorf** [comment]
Otherwise lgtm and passing tests :+1:

**Review by matthias-wright — APPROVED** [review]
LGTM

**Review by ozwaldorf — APPROVED** [review]
Lgtm

**Pull request merged**